x-pack/plugins/lists/server/services/exception_lists/exception_list_client.ts

```diff
@@ -702,13 +705,36 @@ export class ExceptionListClient {
   }: ImportExceptionListAndItemsOptions): Promise<ImportExceptionsResponseSchema> => {
     const { savedObjectsClient, user } = this;
 
-    return importExceptionsAsStream({
+    // validation of import and sorting of lists and items
+    const readStream = createExceptionsStreamFromNdjson(maxExceptionsImportSize);
+    const [parsedObjects] = await createPromiseFromStreams<PromiseFromStreams[]>([
       exceptionsToImport,
-      maxExceptionsImportSize,
+      ...readStream,
+    ]);
+
+    if (this.enableServerExtensionPoints) {
+      await this.serverExtensionsClient.pipeRun(
+        'exceptionsListPreImport',
+        parsedObjects,
+        this.getServerExtensionCallbackContext()
+      );
+    }
+
+    return importExceptions({
+      exceptions: parsedObjects,
       overwrite,
       savedObjectsClient,
       user,
     });
+
+    // FIXME:PT Should the `importExceptionsAsStream()` just be deleted? since it is now not used anywhere?
+    // return importExceptionsAsStream({
```

Comment:
Should I just delete this function? it seems to only be used here.

Also,
Note that the approach would be the same for the client's `importExceptionListAndItemsAsArray()` method below on line 747.